leakcanary-android-core/src/main/java/leakcanary/internal/HeapDumpTrigger.kt

```diff
@@ -55,6 +56,10 @@ internal class HeapDumpTrigger(
     dismissNoRetainedOnTapNotification()
   }
 
+  private val scheduleDismissHeapDumpDisabledNotification = {
```

Comment:
This is not safe. Every call to post/removeCallbacks function will create a new instance of a class that wraps this lambda. It means that a different instance will be passed each time. This is nicely explained in this post: https://medium.com/@LubosMudrak/its-not-the-runnable-you-re-looking-for-321da0fa8e97

tldr; change it to `private val scheduleDismissHeapDumpDisabledNotification = Runnable {`

Reply:
Is this lambda ever posted? I can it being used in `removeCallbacks` but I don't see it being posted anywhere.